readme.md

```diff
@@ -33,20 +33,120 @@ Configuration specifying all parameters:
 docker repository, run the `dockerPush` task.
 
+
+Generating docker-compose dependencies
+--------------------------------------
+
+The plugin provides mechanisms for managing dependencies between docker images
+used in orchestrating `docker-compose` environments. Each project can declare
+its dependencies on other docker images and publish an artifact advertising those
+dependencies. The plugin uses standard Maven/Ivy machanism for declaring and
+resolving dependencies.
+
+The `generateDockerCompose` task generates a `docker-compose.yml` file from
+a user-defined template by replacing each version variable by the concrete version
+declared by the transitive dependencies of the docker configuration.
+
+### Specifying dependencies on Docker images
+The plugin adds a `docker` Gradle component and a `docker` Gradle configuration that
+can be used to specify and publish dependencies on other Docker containers.
+The `generateDockerCompose` task (see below) uses the transitive dependencies
+of the `docker` configuration to populate a `docker-compose.yml.template` file
+with the image versions specified by this project and all its transitive
+dependencies.
+
+    dependencies {
+        docker 'foogroup:barmodule:0.1.2'
+        docker project(":someSubProject")
+    }
+
+    publishing {
+        publications {
+            dockerPublication(MavenPublication) {
+                from components.docker
+                artifactId project.name + "-docker"
+            }
+        }
+    }
+
+The above configuration adds a Maven publication that specifies dependencies
+on `barmodule` and the `someSubProject` Gradle sub project. The resulting POM
+file has two `dependency` entries, one for each dependency.
+
+### Generating docker-compose.yml files from templates
+The `generateDockerCompose` task performs two operations: First, it generates
+a mapping `group:name --> version` from the dependencies of the `docker`
+configuration (see above). Second, it replaces all occurrences of version
+variables of the form `{{group:name}}` in the `docker-compose.yml.template` file
+by the resolved versions and writes the resulting file as `docker-compose.yml`.
+
+** Example **
+
+Assume a `docker-compose.yml.template` as follows:
+
+    myservice:
+      image: 'repository/myservice:latest'
+    otherservice:
+      image: 'repository/otherservice:{{othergroup:otherservice}}'
+
+The `build.gradle` of this project publishes a docker Maven artifact declaring
+a dependency on a docker image published as 'othergroup:otherservice' in version
+0.1.2:
+
+    plugins {
+        id 'maven-publish'
+        id 'com.palantir.docker'
+    }
+
+    docker {
+        name 'foo'
+    }
+
+    group 'mygroup'
+    name 'myservice'
+    version '2.3.4'
+
+    dependencies {
+        docker 'othergroup:otherservice:0.1.2'
+    }
+
+    publishing {
+        publications {
+            dockerPublication(MavenPublication) {
```

Comment:
So it turns out the "group = registryhost, module = name, version = tag" doesn't work since maven artifact ids are restricted to [A-Za-z0-9_\-.]+)  :(